### PR TITLE
Rewrite activity in vue

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

This is the first of a hopefully series of PRs aimed at rewriting parts MPW in Vue.js

## Why vue.js?

Vue is a fairly light framework, and can be integrated into our current codebase from the botton up, meaning that we can start by rewriting singular components, then entire tabs and maybe the entire page eventually.
The syntax is very similar to old school web development, but it avoids a lot of the boilerplate.
For instance, take this snippet of code, repeated many times in our codebase:
```js
document.getElementById('id').style.display = variable ? 'block' : 'none';
```
In vue, this becomes
```html
<div v-if="variable"> </div>
```
Note that we no longer need to assign an id to the div.
Other benefits include:
- The ability to split code in multiple files (index.template.html no longer needs to include everything).
- Harder to include vulnerabilities such as Xss
- Ability to containerize css (and js) to the component, and avoiding having big namespaces
- Probably more?

---

## Why start with the activity?

The activity list is a fairly complex component, which is reused two times (Dashboard and stake tab). It is a perfect example on how other parts can be rewritten as it needs to fetch transaction from the network, process the information and display it.

---

## Testing
- Test that the activity and stake reward list matches with [app.mypivxwallet.org](https://app.mypivxwallet.org/).
- Check that both of the lists are changed upon translation.
- Check that both of the lists update properly when Translations are changed.

---

### CODE REVIEW

**Review by JSKitty — APPROVED** [review]
I've read this implementation over a few times now, and tested lightly as well.

Honestly: I think this is a **great** idea, it looks plenty flexible and easy enough for us to learn and adapt to, it chips away at our huge `global.js` mess, and it reduces the HTML template size a ton, this is definitely the modularisation we needed, in my opinion.

Onwards to Vue.js, I say...

A bug note: the Staking Activity exclusively is not loading for me, and when attempting to load it, my Firefox is freezing up - no errors logged, maybe a bad loop:
image

File: scripts/Activity.vue [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+    // For Staking: Filter the list for only Stakes, display total rewards from known history
+    if (props.rewards) {
+        const arrStakes = arrTXs.filter(
+            (a) => a.type === HistoricalTxType.STAKE
+        );
+
+        if (arrStakes.length !== txs.length) {
+            const nRewards = arrStakes.reduce((a, b) => a + b.amount, 0);
+            console.log(arrStakes);
```
Comment (Liquid369) [inline scripts/Activity.vue:43]: Do we still want/need this log?

**Review by Liquid369 — COMMENTED** [review]
Overall it's looking good, one nit if we really need that `console.log` I commented on.
It runs very well for my large testnet wallet, loading quickly and not freezing up on me gg

File: scripts/Activity.vue (hunk 1 above)
Comment (Duddino) [reply scripts/Activity.vue]: Removed

**Review by Liquid369 — DISMISSED** [review]
tACK 8393b783a7337c920c91c5a9b5e37c3a2e45e228

I am excited about the Vue work! GG @Duddino

**Review by JSKitty — CHANGES_REQUESTED** [review]
There's one minor regression compared to `app.mypivxwallet.org`: After loading the Activity on the Dashboard automatically, the 'Stake History' is not rendered, so when the user navigates to the Staking tab, it will *always* be empty unless you manually hit "Load More" and sync additional history data (more than necessary) from the network.

Previously, MPW renders both at the same time, to prevent this edgecase.

File: scripts/Activity.vue [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+const explorerUrl = computed(() => getNetwork().strUrl);
+
+async function update(fNewOnly = false) {
+    const cNet = getNetwork();
+
+    // Prevent the user from spamming refreshes
+    if (cNet.historySyncing) return;
+
+    // Remember how much history we had previously
+    const nPrevHistory = cNet.arrTxHistory.length;
```
Comment (JSKitty) [inline scripts/Activity.vue:26]: Unused variable

**Review by panleone — APPROVED** [review]
tACK 2eec65e079d0624908831ea3ea4e3e72e0f3e06f
I really like this PR: apart from the activity refactor, with vue we can finally unify UI and logic in a single file instead of putting the UI in global.js Also I like lines like `export const translation = reactive({});` in such a way that translation is automatically updated.

File: scripts/Activity.vue [hunk 3]
```diff
@@ -61,19 +61,20 @@ async function update(fNewOnly = false, sync = true) {
...
+	
+	if (txCount !== cNet.arrTxHistory.length || !sync) {
+            arrTXs = cNet.arrTxHistory;
+	} else {
+            arrTXs = await cNet.syncTxHistoryChunk(fNewOnly);
+	}
+	txCount = arrTXs.length;
+	if (!arrTXs || cNet !== getNetwork()) return;
+    } finally {
+	updating.value = false;
```
Comment (JSKitty) [inline scripts/Activity.vue:76]: I think this section should either be commented a bit, or cleaned somehow, it's a bit 'messy' looking and takes a moment to unwrap.

**Review by JSKitty — COMMENTED** [review]
`npm run prettier`

File: scripts/Activity.vue (hunk 3 above)
Comment (Duddino) [reply scripts/Activity.vue]: Done

**Review by panleone — DISMISSED** [review]
re-tACK 00e1e70b71f126a215a4f06dc34ba13c7a042e38

**Review by JSKitty — DISMISSED** [review]
**tACK** 76a82667e0dd157e8ca4d6160012e5aedcf854f9

Smooth, almost identical functionality to Production, a really good cleanup for `global.js`, and a step towards MPW modularity

**Review by JSKitty — APPROVED** [review]
**tACK** 190b0446c717b9082b906681aa9f97b6b61a08d7

**Review by Liquid369 — APPROVED** [review]
tACK 190b0446c717b9082b906681aa9f97b6b61a08d7

**Review by panleone — APPROVED** [review]
tACK 190b0446c717b9082b906681aa9f97b6b61a08d7

**Pull request merged**